src/gluonts/model/npts/_model.py

```diff
@@ -154,7 +154,12 @@ def predict(
         # Forecast takes as input the prediction range samples, the start date
         # of the prediction range, and the frequency of the time series.
         samples_pred_range = samples[:, train_length:]  # prediction range only
-        forecast_start = targets.index[-1] + 1
+
+        # For freq M and W pandas seems to lose the freq of the timestamp,
+        # so we explicitly set it.
+        forecast_start = pd.Timestamp(
+            targets.index[-1] + 1 * targets.index.freq, freq=targets.index.freq
```

Comment:
Doesn't have same ring to it :|

Reply:
This will be removed with the latest pandas, see #177